### ISSUE TITLE
highlight expressions on hover

### ISSUE THREAD

**Issue by freebroccolo** [freebroccolo/vscode-reasonml#8]
The current default behavior is to show subtle highlighting of identifiers on hover. However, this doesn't really mesh well with the `position` info which is supposed to indicate whether an expression is in a tail (call) position.

What happens is that when you hover over the head term of an application, it will report `position: normal`, which is fine, but a little bit surprising. You only see the `position: tail` if you hover slightly in front of the head term but now you don't see anything highlighted in the buffer because there is no identifier under the cursor. What *should* happen is the entire application should be highlighted.

tldr: the highlighting extents on hover should be calculated by merlin from the shape of the expression at the cursor position rather than being calculated by the editor based on identifiers.

**Comment by freebroccolo** [comment]
Related to #60.
